daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/ContractStateMachine.scala

```diff
@@ -125,13 +125,12 @@ class ContractStateMachine[Nid](mode: ContractKeyUniquenessMode) {
       }
 
     /** Visit a create node */
-    def handleCreate(node: Node.Create): Either[KeyInputError, State] =
-      visitCreate(node.templateId, node.coid, node.keyValue).left.map(Right(_))
+    def handleCreate(node: Node.Create, key: Option[GlobalKey]): Either[KeyInputError, State] =
+      visitCreate(node.coid, key).left.map(Right(_))
 
     private[lf] def visitCreate(
-        templateId: TypeConName,
         contractId: ContractId,
-        mbKey: Option[Value],
+        mbKey: Option[GlobalKey],
```

Comment:
Likewise, a nice refactor/simplification using the GlobalKey directly (especially as it already carries the templateID).

Reply:
There is two additional advantage: 
- GlobalKey contains the hash of the key, so we do not need to recompute it
- Since we have computed the hash of the key, we no the key is valid (i.e. it does not contain contract IDs)